From fce72209cb4d0e43a9c696325cb94f78d0f5038a Mon Sep 17 00:00:00 2001
From: Antheas Kapenekakis <lkml@antheas.dev>
Date: Thu, 17 Apr 2025 11:41:53 +0200
Subject: [PATCH 050/112] Revert "platform/x86: firmware_attributes_class: Drop
 lifecycle functions"

This reverts commit 6af39604c788d9df0ca697bab603202cf74c1a8c.
---
 drivers/platform/x86/firmware_attributes_class.c | 13 +++++++++++++
 drivers/platform/x86/firmware_attributes_class.h |  2 ++
 2 files changed, 15 insertions(+)

diff --git a/drivers/platform/x86/firmware_attributes_class.c b/drivers/platform/x86/firmware_attributes_class.c
index 736e96c18..87672c49e 100644
--- a/drivers/platform/x86/firmware_attributes_class.c
+++ b/drivers/platform/x86/firmware_attributes_class.c
@@ -22,6 +22,19 @@ static __exit void fw_attributes_class_exit(void)
 }
 module_exit(fw_attributes_class_exit);
 
+int fw_attributes_class_get(const struct class **fw_attr_class)
+{
+	*fw_attr_class = &firmware_attributes_class;
+	return 0;
+}
+EXPORT_SYMBOL_GPL(fw_attributes_class_get);
+
+int fw_attributes_class_put(void)
+{
+	return 0;
+}
+EXPORT_SYMBOL_GPL(fw_attributes_class_put);
+
 MODULE_AUTHOR("Mark Pearson <markpearson@lenovo.com>");
 MODULE_DESCRIPTION("Firmware attributes class helper module");
 MODULE_LICENSE("GPL");
diff --git a/drivers/platform/x86/firmware_attributes_class.h b/drivers/platform/x86/firmware_attributes_class.h
index d27abe54f..ef6c3764a 100644
--- a/drivers/platform/x86/firmware_attributes_class.h
+++ b/drivers/platform/x86/firmware_attributes_class.h
@@ -8,5 +8,7 @@
 #include <linux/device/class.h>
 
 extern const struct class firmware_attributes_class;
+int fw_attributes_class_get(const struct class **fw_attr_class);
+int fw_attributes_class_put(void);
 
 #endif /* FW_ATTR_CLASS_H */
-- 
2.52.0

